nixos/modules/services/networking/seafile.nix

```diff
@@ -162,32 +245,38 @@ in
           ProtectKernelLogs = true;
           ProtectControlGroups = true;
           RestrictNamespaces = true;
+          RemoveIPC = true;
           LockPersonality = true;
           RestrictRealtime = true;
           RestrictSUIDSGID = true;
+          NoNewPrivileges = true;
           MemoryDenyWriteExecute = true;
           SystemCallArchitectures = "native";
           RestrictAddressFamilies = [ "AF_UNIX" "AF_INET" ];
         };
+
+        dataDir = if (!builtins.isNull cfg.dataDir) then cfg.dataDir else "${seafRoot}/data";
+        optionalDataDir = lib.lists.optional (!builtins.isNull cfg.dataDir) cfg.dataDir;
       in
       {
         seaf-server = {
           description = "Seafile server";
           partOf = [ "seafile.target" ];
-          after = [ "network.target" ];
+          unitConfig.RequiresMountsFor = optionalDataDir;
+          requires = [ "mysql.service" ];
```

Comment:
you could use a new option as mentioned above here to only conditionally require this if it's running locally